app/forms/date_form.py

```diff
@@ -84,3 +98,62 @@ class MonthYearDateForm(Form):
     MonthYearDateForm.month = SelectField(choices=month_choices, default='', validators=validate_with)
 
     return MonthYearDateForm
+
+
+def get_min_max_dates_for_single_date_period_validation(answer, answer_store, metadata):
+    """
+    Gets attributes within a minimum or maximum of a date field and validates that the entered date
+    is valid.
+
+    :param answer: The answer which contains the minimum or maximum
+    :param answer_store: The current answer store
+    :param metadata: metadata for reference meta dates
+    :return: attributes
+    """
+    minimum_referenced_date, maximum_referenced_date = None, None
+
+    if 'minimum' in answer:
+        minimum_referenced_date = get_referenced_offset_value(answer['minimum'], answer_store, metadata)
+    if 'maximum' in answer:
+        maximum_referenced_date = get_referenced_offset_value(answer['maximum'], answer_store, metadata)
+
+    if minimum_referenced_date and maximum_referenced_date:
+        if minimum_referenced_date > maximum_referenced_date:
+            raise Exception('The minimum offset date is greater than the maximum offset date for {}.'.format(answer['id']))
+
+    return minimum_referenced_date, maximum_referenced_date
+
+
+def get_referenced_offset_value(answer_min_or_max, answer_store, metadata):
+    """
+    Gets value of the referenced date type, whether it is a value,
+    id of an answer or a meta date. Then adds/subtracts offset from that value and returns
+    the new offset value
+
+    :param answer_min_or_max: The minimum or maximum object which contains
+    the referenced value.
+    :param answer_store: The current answer store
+    :param metadata: metadata for reference meta dates
+    :return: value
+    """
+    value = None
+
+    if 'value' in answer_min_or_max:
+        if answer_min_or_max['value'] == 'now':
+            value = datetime.utcnow().strftime('%Y-%m-%d')
+        else:
+            value = answer_min_or_max['value']
+    elif 'meta' in answer_min_or_max:
+        value = get_metadata_value(metadata, answer_min_or_max['meta'])
+    elif 'answer_id' in answer_min_or_max:
+        value = get_answer_store_value(answer_min_or_max['answer_id'], answer_store, group_instance=0)
```

Comment:
This would fail in a repeating group wouldn't it? As it would always validate against the value of the answer in the first group instance, not the current group instance.

Reply:
Ticket raised in planning backlog to address this, ass per Davids request

Reply:
@nealedj, @shahi645. Why are we introducing tech debt? In my opinion if we know this feature will fail in certain conditions it shouldn't be put in. I wouldn't class this as an edge case as we already know we're likely to need this in repeating groups for both Census and LFS.

Reply:
Having spoken with @nealedj, I realise this is the same format as used by numbers so a separate PR will be needed for that either way. To get this PR through then i'm happy for this to be addressed at the same time and in the same fashion as will be needed for numbers.